server/lease/lessor.go

```diff
@@ -351,6 +366,10 @@ func (le *lessor) Checkpoint(id LeaseID, remainingTTL int64) error {
 	if l, ok := le.leaseMap[id]; ok {
 		// when checkpointing, we only update the remainingTTL, Promote is responsible for applying this to lease expiry
 		l.remainingTTL = remainingTTL
+		cv := le.cluster.Version()
+		if le.checkpointPersist || (cv != nil && greaterOrEqual(*cv, v3_6)) {
```

Comment:
nit: shell we have this embedded in a getter on `le`. 

le.isCheckpointPersist() that hides the v3.6 resolution ?

Reply:
SG